### PR TITLE
Support unions as return types

### PR DESCRIPTION
**Pull request by newhouse**
Adds support for UNIONs as return types to Queries and Mutations.

### CODE REVIEW

File: lib/reducer.ts [hunk 1]
```diff
@@ -112,25 +112,28 @@ export const resolveDefaultValue = (curr: any) => {
   // Dig out the underlying type in case it's a LIST or a NON_NULL or both
   while (true) {
     if (isIntrospectionListTypeRef(type)) {
-        isList = true
```
Comment (newhouse) [inline lib/reducer.ts:115]: Prettier...

File: lib/typeGuards.ts [hunk 2]
```diff
@@ -18,6 +18,25 @@ import {
...
+const KIND_ENUM = 'ENUM'
+const KIND_UNION = 'UNION'
+
+export const SUPPORTED_KINDS = [
+  KIND_SCALAR,
+  KIND_OBJECT,
+  KIND_INPUT_OBJECT,
+  KIND_INTERFACE,
+  KIND_ENUM,
+  KIND_UNION,
```
Comment (newhouse) [inline lib/typeGuards.ts:37]: Fix was adding `UNION` to the array that gets checked...

File: lib/typesMapping.ts [hunk 3]
```diff
@@ -55,12 +56,7 @@ export const graphqlToJSONType = (
 
     const jsonType = {} as JSONSchema6
 
-    if (
-      includes(
-        ['OBJECT', 'INPUT_OBJECT', 'INTERFACE', 'ENUM', 'SCALAR'],
-        k.kind
-      )
-    ) {
+    if (includes(SUPPORTED_KINDS, k.kind)) {
```
Comment (newhouse) [inline lib/typesMapping.ts:59]: Fix was adding `UNION` to the array that gets checked...

**Comment by newhouse** [comment]
Bump @charlypoly

**Comment by newhouse** [comment]
Bump, bump @charlypoly

File: lib/typeGuards.ts [hunk 4]
```diff
@@ -18,6 +18,25 @@ import {
 } from 'graphql'
 import { filter, has, startsWith, includes } from 'lodash'
 
+const KIND_LIST = 'LIST'
```
[the remark this answers is not in the document]
Comment (newhouse) [reply lib/typeGuards.ts]: Done! Ready for re-review.

**Comment by newhouse** [comment]
bump @charlypoly

**Comment by charlypoly** [comment]
@newhouse this PR has conflicts with the `json5` one that just got merged.

I tried to push on your branch resolution of the conflicts, but I don't have the access rights.

Could you resolve those conflicts?

**Comment by newhouse** [comment]
@charlypoly done!

Can you let me know if/when this and the other PR have been published so I can update `spectaql` to use your library (instead of my fork) again?  Thanks!

**Comment by charlypoly** [comment]
Published in `graphql-2-json-schema@0.9.1`

**Pull request merged**